gradle-failure-reports/src/main/java/com/palantir/gradle/failurereports/FailureReportsRootPlugin.java

```diff
@@ -44,52 +39,33 @@ public void apply(Project project) {
         }
         FailureReportsExtension failureReportsExtension =
-                project.getExtensions().create("failureReports", FailureReportsExtension.class);
-        TaskProvider<FinalizerTask> finalizerTask = project.getTasks()
-                .register(FINALIZER_TASK, FinalizerTask.class, task -> {
-                    task.getOutputFile().set(failureReportsExtension.getFailureReportOutputFile());
-                });
-        project.getRootProject()
-                .allprojects(subProject -> subProject.getPlugins().apply(FailureReportsProjectsPlugin.class));
-        collectVerifyLocksFailureReports(project, finalizerTask);
-        collectOtherTaskFailures(project, finalizerTask);
-    }
+                ExtensionUtils.maybeCreate(project, "failureReports", FailureReportsExtension.class);
+        CompileFailuresService.getSharedCompileFailuresService(project, failureReportsExtension);
 
-    private void collectVerifyLocksFailureReports(Project project, TaskProvider<FinalizerTask> finalizerTask) {
-        project.getPluginManager().withPlugin("com.palantir.versions-lock", _javaPlugin -> {
-            TaskProvider<Task> verifyLocksTask = project.getTasks().named(VERIFY_LOCKS_TASK);
-            verifyLocksTask.configure(task -> task.finalizedBy(finalizerTask));
+        project.allprojects(subproject -> subproject.getPluginManager().apply(FailureReportsProjectsPlugin.class));
+        if (GradleVersion.version(project.getGradle().getGradleVersion()).compareTo(GRADLE_FLOW_ACTIONS_ENABLED) >= 0) {
+            project.getPluginManager().apply(FailureReportsFlowActionsPlugin.class);
+        } else {
+            project.getGradle().addBuildListener(new BuildListener() {
+                @Override
+                public void settingsEvaluated(Settings _settings) {}
 
-            finalizerTask.configure(finalizer -> finalizer.getFailureReports().addAll(project.provider(() -> {
-                if (FailureReporterResources.executedAndFailed(verifyLocksTask.get())
-                        && ThrowableFailureReporter.maybeGetFailureReport(verifyLocksTask.get())
-                                .isEmpty()) {
-                    return List.of(VerifyLocksFailureReporter.getFailureReport(verifyLocksTask.get()));
-                }
-                return List.of();
-            })));
-        });
-    }
+                @Override
+                public void projectsLoaded(Gradle _gradle) {}
+
+                @Override
+                public void projectsEvaluated(Gradle _gradle) {}
 
-    private void collectOtherTaskFailures(Project project, TaskProvider<FinalizerTask> finalizerTask) {
-        project.allprojects(subProject -> {
-            TaskContainer projectTasks = subProject.getTasks();
-            projectTasks.configureEach(task -> {
-                if (isAllowedTask(task)) {
-                    task.finalizedBy(finalizerTask);
+                @Override
+                public void buildFinished(BuildResult result) {
```

Comment:
I couldn't find any way to send the buildService here in the buildListener, this is why I added the AutoClosable.

from docs (https://docs.gradle.org/current/userguide/build_services.html#implementing_a_build_service):
`A build service implementation can also optionally implement AutoCloseable, in which case Gradle will call the build service instance’s close() method when it discards the service instance. This happens some time between completion of the last task that uses the build service and the end of the build.`

Reply:
Does it not work if you just pass it into this anon class?